keytool/src/main/scala/org/constellation/keytool/DeterministicKeyGen.scala

```diff
@@ -0,0 +1,43 @@
+package org.constellation.keytool
+
+import com.typesafe.scalalogging.StrictLogging
+
+import java.nio.charset.Charset
+import java.security.spec.ECGenParameterSpec
+import java.security.{SecureRandom, _}
+
+object DeterministicKeyGen extends StrictLogging {
+
+  import KeyUtils._
+
+  private val letters: List[Char] = ('a' to 'z').toList
+  // A -> 1 ; Z -> 26
+  val letterMapping: Map[Char, Int] = letters.zipWithIndex.toMap.mapValues(_ + 1)
+
+  def mapCompoundLetters(seed: String): String = {
+    seed.toLowerCase.filter(letters.contains).flatMap(letterMapping.get).mkString("")
+  }
+
+  def makeKeyPairsLetterMap(count: Int = 10, seed: String): List[KeyPair] = {
+    val seedBytes = BigInt(mapCompoundLetters(seed)).toByteArray
+    makeKeyPairs(count, seedBytes)
+  }
+
+  def makeKeyPairsUTFBytes(count: Int = 10, seed: String): List[KeyPair] = {
+    makeKeyPairs(count, seed.getBytes(Charset.forName("UTF-8")))
+  }
+
+  // TODO: SHA1PRNG requires higher byte counts for seed security -- need to expand original bytes and pad
+  def makeKeyPairs(count: Int = 10, seed: Array[Byte]): List[KeyPair] = {
```

Comment:
This function isn't pure, so its execution should be delayed in a monad. Please take a look how it looks in other parts of the codebase and make it consistent.

Reply:
Will update, just based this on prior functions -- I saw wrappers around them elsewhere but just copied from keytool

Reply:
do we have unwrapped side effects in keytool? can you point me it cause I cannot recall